src/inmanta/agent/handler.py

```diff
@@ -221,6 +231,20 @@ def _log_msg(
         raise NotImplementedError
 
 
+class ContextResourceState(str, Enum):
+    """
+    Used by the HandlerContext to determine the state of the resource
+    Similar to const.ResourceState but with some states removed and with the added skipped_for_dependency state
+    """
+
+    skipped = "skipped"
+    deployed = "deployed"
+    failed = "failed"
+    dry = "dry"
+    skipped_for_dependency = "skipped_for_dependency"
+    unavailable = "unavailable"  # to be removed?
```

Comment:
Yes, It was supposed to be a question to the reviewers. My thought process is that this will be caught and answered by the reviewers and then removed. But, like you mentioned, it is very likely that comments like this get lost and pollute the code. I'll start adding FIXMEs instead. (Wouter answered in another thread)